src/scaler/spreadscaler/mod.rs

```diff
@@ -394,10 +400,46 @@ fn compute_spread(spread_config: &SpreadScalerProperty) -> Vec<(Spread, usize)>
         // there was a case that I didn't imagine
         Ordering::Greater => {
             warn!("Requesting more actor instances than were specified");
-            spreads
+            computed_spreads
         }
-        Ordering::Equal => spreads,
+        Ordering::Equal => computed_spreads,
+    };
+
+    // case when spread_config.spread.is_empty() <=> requested_spreads.is_empty()
+    if computed_spreads.len() == 1 {
+        return computed_spreads;
+    }
+
+    // this is an invariant check; the predicate should never be true.
+    // this check implicitly ensures that the reordering logic below is safe with regards to indexing.
+    if computed_spreads.len() != requested_spreads.len() {
+        panic!("Computed spreads and requested spreads are not the same length. This should not happen");
     }
+
+    // spreads are returned in the order that they were declared in the config (ie, not by weight).
+    // this shouldn't matter; however, it is a safe behavior with respect to downstream processing.
+    let requested_spreads_indices: HashMap<String, usize> = spread_config
+        .spread
+        .iter()
+        .enumerate()
+        .map(|(idx, spread)| (spread.name.to_owned(), idx))
+        .collect();
+
+    computed_spreads.iter().fold(
+        vec![(Spread::default(), 0); computed_spreads.len()],
+        |mut output, spread| {
+            if let Some(idx) = requested_spreads_indices.get(&spread.0.name) {
+                *output.index_mut(*idx) = spread.to_owned()
+            } else {
+                panic!(
+                    "Spread(name={}) was not found in spread_config (by way of requested_spreads_indices). This should not happen",
+                    spread.0.name
+                );
```

Comment:
This seems impossible to reach, but I would still rather this be an error log than an explicit `panic` to avoid crashing our process/task in an extreme edge case. FWIW, I avoid any panics/unwraps outside of tests because I never want to see a process crash.

I still feel this bit of logic isn't necessary since this is an internal function to the spreadscaler itself, is your concern that we might want to read or examine the computed spreads to match it up to the YAML here and, when we sort by weight, it might not be the same order as it's specified in the manifest?

Reply:
yes, in a nutshell, that is my concern. basically, not knowing enough about the entire codebase, i guess i am somewhat wary of subtle bugs that can emerge from other downstream operations that implicitly/unintentionally rely on the order in the YAML.

Now, taking a step back, I am also realizing that your understanding/experience with the codebase should be enough of a reason for me downgrade my concern if you believe returning the list sorted by weight should be fine. With that said, I will go ahead and remove the reordering logic. Thanks for feedback on this by the way.

Reply:
@ahmedtadde This part of your logic showed some real thought behind it and I really appreciate this part of the contribution. Though it's not present today, feel free to leave a comment noting the change in behavior here and that it would need to be revisited if we depend on a specific order in the future. 

Once your next commit comes in removing this and the previous possible (I mean, probably impossible but just presence of) `panic`, this should be good to go!